File: Day02/go/part01.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strconv"
	"strings"
)

func passwordValidator() int {

	f, err := os.Open("./blob.txt")
	if err != nil {
		log.Fatal(err)
	}
	defer f.Close()

	scanner := bufio.NewScanner(f)
	total := 0
	for scanner.Scan() {
		text := strings.Fields(scanner.Text())
		params := text[0]
		min, err := strconv.Atoi(strings.Split(params, "-")[0])
		max, err := strconv.Atoi(strings.Split(params, "-")[1])
		char := strings.Replace(text[1], ":", "", -1)
		password := text[len(text)-1]
		if strings.Contains(password, char) && min <= strings.Count(password, char) && strings.Count(password, char) <= max {
			total++
		}

		if err != nil {
			log.Fatal(err)
		}
	}
	return total

}

func main() {
	fmt.Println(passwordValidator())
}

File: Day02/part01.py
def password_validator(file: str) -> int:
    with open(file, "r") as f:
        total = 0
        for line in f.readlines():
            ln = line.split()
            params = ln[0]
            min_ = int(params.split("-")[0])
            max_ = int(params.split("-")[-1])
            char = ln[1].replace(":", "")
            password = ln[-1]
            if char in password and min_ <= password.count(char) <= max_:
                total += 1
    return total


if __name__ == "__main__":
    print(password_validator("./blob.txt"))
